modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java

```diff
@@ -51,52 +46,65 @@
 
+        DeviceIdentifier deviceIdentifier = AndroidAPIUtils.convertToDeviceIdentifierObject(id);
         try {
-            if (resultOperations != null) {
+            if (!AndroidDeviceUtils.isValidDeviceIdentifier(deviceIdentifier)) {
+                String errorMessage = "Device not found for identifier '" + id + "'";
+                message = Message.responseMessage(errorMessage).
+                        responseCode(Response.Status.BAD_REQUEST.toString()).build();
+                log.error(errorMessage);
+                throw new AndroidOperationException(message, responseMediaType);
+            }
+            if (log.isDebugEnabled()) {
+                log.debug("Invoking Android pending operations:" + id);
+            }
+            if (resultOperations != null && !resultOperations.isEmpty()) {
                 updateOperations(id, resultOperations);
             }
         } catch (OperationManagementException e) {
-            message.setResponseMessage("Issue in retrieving operation management service instance");
-            log.error(message.getResponseMessage(), e);
+            log.error("Issue in retrieving operation management service instance", e);
+        } catch (PolicyComplianceException e) {
+            log.error("Issue in updating Monitoring operation");
+        } catch (DeviceManagementException e) {
+            log.error("Issue in retrieving device management service instance", e);
         }
 
-        DeviceIdentifier deviceIdentifier = AndroidAPIUtils.convertToDeviceIdentifierObject(id);
-        List<? extends org.wso2.carbon.device.mgt.common.operation.mgt.Operation> operations;
-        List<Operation> pendingOperations = new ArrayList<Operation>();
+        List<? extends Operation> pendingOperations;
         try {
-            operations = AndroidAPIUtils.getPendingOperations(deviceIdentifier);
-            for(Operation operation:operations){
-                pendingOperations.add(operation);
-            }
+            pendingOperations = AndroidAPIUtils.getPendingOperations(deviceIdentifier);
         } catch (OperationManagementException e) {
             String errorMessage = "Issue in retrieving operation management service instance";
-            message.setResponseMessage(errorMessage);
-            message.setResponseCode(Response.Status.INTERNAL_SERVER_ERROR.toString());
+            message = Message.responseMessage(errorMessage).
+                    responseCode(Response.Status.INTERNAL_SERVER_ERROR.toString()).build();
             log.error(errorMessage, e);
             throw new AndroidOperationException(message, responseMediaType);
         }
-
         return pendingOperations;
     }
 
     @POST
     @Produces(MediaType.APPLICATION_JSON)
     @Path("lock")
-    public Response configureDeviceLock(@HeaderParam("Accept") String acceptHeader, List<String> deviceIDs) {
+    public Response configureDeviceLock(@HeaderParam(ACCEPT) String acceptHeader, List<String> deviceIDs) {
```

Comment:
While we allow any type of response via the accept header,  why do we also limit response to JSON?